### ISSUE TITLE
salt-cloud throw out traceback info when cloud profile is not exist

### ISSUE THREAD

**Issue by yi9** [saltstack/salt#28683]
If cloud profile id in cloud map file, is not matching any existing profile, when build or delete a vm, salt-cloud won't exit friendly, it throws out traceback information. as below.

Nitin has been notified. @nmadhok
```
root@devccitmgt2:/usr/lib/python2.6/site-packages/salt/cloud# salt-cloud -d -m splunk-yi9-profiletest.map
[INFO    ] salt-cloud starting
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
SaltCloudNotFound: The specified map file does not exist: splunk-yi9-profiletest.map

Traceback (most recent call last):
  File "/usr/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 316, in salt_cloud
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cloud/cli.py", line 86, in run
    mapper = salt.cloud.Map(self.config)
  File "/usr/lib/python2.6/site-packages/salt/cloud/__init__.py", line 1558, in __init__
    self.rendered_map = self.read()
  File "/usr/lib/python2.6/site-packages/salt/cloud/__init__.py", line 1646, in read
    self.opts['map']
SaltCloudNotFound: The specified map file does not exist: splunk-yi9-profiletest.map
Traceback (most recent call last):
  File "/usr/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 316, in salt_cloud
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cloud/cli.py", line 86, in run
    mapper = salt.cloud.Map(self.config)
  File "/usr/lib/python2.6/site-packages/salt/cloud/__init__.py", line 1558, in __init__
    self.rendered_map = self.read()
  File "/usr/lib/python2.6/site-packages/salt/cloud/__init__.py", line 1646, in read
    self.opts['map']
salt.exceptions.SaltCloudNotFound: The specified map file does not exist: splunk-yi9-profiletest.map
```

**Comment by rallytime** [comment]
@yi9 We should definitely be cleaning up those types of nasty stacktraces! We'll get this cleaned up. However, can you also post the output of `salt-cloud --versions-report`, please?

**Comment by rallytime** [comment]
This has been fixed on the 2015.5 branch. I was able to reproduce the error on 2014.7 like so:
```
# salt-cloud -d -m /some/map/file
[INFO    ] salt-cloud starting
[WARNING ] The required 'api_key' configuration setting is missing on the 'digital_ocean' driver(under the 'digital_ocean' alias)
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
SaltCloudNotFound: The specified map file does not exist: /some/map/file

Traceback (most recent call last):
  File "/usr/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/root/SaltStack/salt/salt/scripts.py", line 210, in salt_cloud
    client.run()
  File "/root/SaltStack/salt/salt/cloud/cli.py", line 83, in run
    mapper = salt.cloud.Map(self.config)
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1512, in __init__
    self.rendered_map = self.read()
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1614, in read
    self.opts['map']
SaltCloudNotFound: The specified map file does not exist: /some/map/file
Traceback (most recent call last):
  File "/usr/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/root/SaltStack/salt/salt/scripts.py", line 210, in salt_cloud
    client.run()
  File "/root/SaltStack/salt/salt/cloud/cli.py", line 83, in run
    mapper = salt.cloud.Map(self.config)
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1512, in __init__
    self.rendered_map = self.read()
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1614, in read
    self.opts['map']
salt.exceptions.SaltCloudNotFound: The specified map file does not exist: /some/map/file
```
But the stacktrace is cleaned up already in 2015.5:
```
# salt-cloud -d -m /some/map/file
[INFO    ] salt-cloud starting
[ERROR   ] The specified map file does not exist: '/some/map/file'
Usage: salt-cloud

salt-cloud: error: There was an error generating the mapper.
```
This was fixed by PR #25877.
